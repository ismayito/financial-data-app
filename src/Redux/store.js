import { configureStore } from '@reduxjs/toolkit';
import FinancialSlice from './FinancialSlice';

const store = configureStore({
  reducer: {
    financial: FinancialSlice,
  },
});

export default store;
