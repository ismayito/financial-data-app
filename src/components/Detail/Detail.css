.detailHeader {
  display: flex;
  justify-content: start;
  background-color: #4369b1;
}

.detailHead {
  color: white;
  margin-top: 0;
  margin-bottom: 0;
  padding-top: 4px;
  padding-bottom: 2px;
}

.detailCardContainer {
  display: flex;
  flex-direction: column;
  color: white;
}

.detailCardContainer :nth-child(odd) {
  background-color: #5687e3;
  opacity: 0.9;
}

.detailCardContainer :nth-child(even) {
  background-color: #5687e3;
}

.detailCard {
  display: flex;
  justify-content: space-between;
  padding: 12px 12px;
  font-family: "Lato", sans-serif;
  font-weight: 900;
}

.cardInfo {
  display: flex;
  align-items: center;
  gap: 4px;
}
