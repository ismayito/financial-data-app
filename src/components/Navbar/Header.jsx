import React from 'react';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import { BsChevronLeft } from 'react-icons/bs';
import { MdKeyboardVoice } from 'react-icons/md';
import { IoIosSettings } from 'react-icons/io';
import '../../pages/Pages.css';

const Header = ({ title }) => {
  const navigate = useNavigate();
  const handleNavigation = () => {
    navigate('/');
  };

  return (
    <div>
      <header>
        <nav className="navBar">
          <section className="logo">
            <button type="submit" className="navigationButton" onClick={handleNavigation}>
              <BsChevronLeft onClick={handleNavigation} />
              <p className="link">Finance</p>
            </button>
          </section>
          <p className="title">{title}</p>
          <section className="navIcons">
            <MdKeyboardVoice />
            <IoIosSettings />
          </section>

        </nav>
      </header>
    </div>
  );
};
Header.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Header;
