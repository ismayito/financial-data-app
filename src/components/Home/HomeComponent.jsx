import React, { useEffect, useState } from 'react';
import { BsArrowRightCircle } from 'react-icons/bs';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { fetchFinance, financialItemDetail } from '../../Redux/FinancialSlice';
import '../../pages/Pages.css';

const HomeComponent = () => {
  const [searchInput, setSearchInput] = useState('');
  const [returnedSearchData, setReturnedSearchData] = useState([]);

  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { financial } = useSelector((store) => store.financial);
  const displayData = financial?.slice(0, 10);

  useEffect(() => {
    dispatch(fetchFinance());
  }, [
    dispatch,
  ]);

  // callback to handle navigation to details page on clicking a single card item
  const handleRoute = (id) => {
    dispatch(financialItemDetail(id));
    navigate('/details');
  };

  // callback to handle search input functionality to details page on clicking a single card item
  const handleSearchInput = (e) => {
    setSearchInput(e.target.value);
    if (searchInput === '') { return; }
    const searchedData = displayData.filter((item) => {
      if (item.exchange.toLowerCase().includes(searchInput.toLowerCase())) {
        return true;
      }
      return false;
    });
    setReturnedSearchData(searchedData);
  };

  return (
    <>
      <div className="headSection">
        <section className="imageContainer">
          <img src="../assets/images/financeimage.webp" alt="fruityImage" className="fruityImage" />
        </section>
        <section className="introInfo">
          <h1 className="heading">
            Finance
            <br />
            <span>
              {displayData.length}
              {' '}
              companies
            </span>
          </h1>
        </section>
      </div>
      <section className="inputContainer">
        <input
          type="search"
          placeholder="search..."
          className="input"
          onChange={handleSearchInput}
        />
      </section>
      <p className="priceTitle">Price By Stock Company</p>
      <div className="cardContainer">
        {
            searchInput === '' ? (displayData.map((data) => (
              <button key={data.symbol} className="button" type="button" onClick={() => handleRoute(data.symbol)}>
                <BsArrowRightCircle className="circle" />
                <p className="exchange">{data.exchange}</p>
                <p className="price">
                  price:
                  {data.price}
                </p>
              </button>
            ))) : (
              returnedSearchData.map((data) => (
                <button key={data.symbol} className="button" type="button" onClick={() => handleRoute(data.symbol)}>
                  <BsArrowRightCircle />
                  <p className="exchange">{data.exchange}</p>
                  <p className="price">
                    price:
                    {data.price}
                  </p>
                </button>
              ))
            )
        }
      </div>
    </>
  );
};

export default HomeComponent;
