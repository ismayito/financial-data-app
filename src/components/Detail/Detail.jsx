import React from 'react';
import { BsArrowRightCircle } from 'react-icons/bs';
import { useSelector } from 'react-redux';
import '../../pages/Pages.css';
import './Detail.css';

const Detail = () => {
  const { financial } = useSelector((store) => store.financial);
  console.log(financial);
  return (
    <div>
      <div className="headSection">
        <section className="imageContainer">
          <img src="../assets/images/detail.jpg" alt="financeImage" className="fruityImage" />
        </section>
        {
          financial.map(item=>(
            <section className="introInfo">
          <h1 className="heading">
            {item.exchange}
          </h1>
        </section>
            ))
        }
      </div>
      {financial.map(item=>(
          <section className="detailHeader">
             <p className="detailHead">{item.name}</p>
            </section>
      ))}
      {financial.map(item=>(
      <div className="detailCardContainer">
      <div className="detailCard">
        <p>Exchange Short Name</p>
        <div className="cardInfo">
          <p>{item.exchangeShortName}</p>
          <BsArrowRightCircle />
        </div>
        </div>
        <div className="detailCard">
        <p>Symbol</p>
        <div className="cardInfo">
          <p>{item.symbol}</p>
          <BsArrowRightCircle />
        </div>
        </div>
        <div className="detailCard">
        <p>type</p>
        <div className="cardInfo">
          <p>{item.type}</p>
          <BsArrowRightCircle />
        </div>
        </div>
        <div className="detailCard">
        <p>Price</p>
        <div className="cardInfo">
          <p>{item.price}</p>
          <BsArrowRightCircle />
        </div>
        </div>
      </div>
      ))}
    </div>
  );
};

export default Detail;
