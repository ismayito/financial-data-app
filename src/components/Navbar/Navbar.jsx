import React from 'react';
import { NavLink } from 'react-router-dom';
import Header from './Header';
import './Component.css';

const Navbar = () => (
  <div className="links">
    <NavLink to="/">
      <Header />
    </NavLink>
    {/* <NavLink to="/details">Details</NavLink> */}
  </div>
);

export default Navbar;
