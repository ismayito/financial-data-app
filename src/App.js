import { Route, Routes } from 'react-router-dom';
import Home from './pages/Home';
import Details from './pages/Details';
import './App.css';
import Layout from './components/Layout';

function App() {
  return (
    <div className="App">
      <Routes>
        <Route path="/" element={<Layout />}>
          <Route index element={<Home />} />
          <Route path="/details" element={<Details />} />
        </Route>
      </Routes>
    </div>
  );
}

export default App;
