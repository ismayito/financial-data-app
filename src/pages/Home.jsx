import React from 'react';
import './Pages.css';
import HomeComponent from '../components/Home/HomeComponent';

const Home = () => (
  <div className="homeContainer">
    <HomeComponent />
  </div>
);

export default Home;
