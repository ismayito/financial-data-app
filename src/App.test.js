import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from './Redux/store';
import Detail from './components/Detail/Detail';
import Header from './components/Navbar/Header';
import Navbar from './components/Navbar/Navbar';
import Layout from './components/Navbar/Layout';
import HomeComponent from './components/Home/HomeComponent';
import Home from './pages/Home';
import App from './App';

describe('Tests for the financial App', () => {
  it('Tests the display of Header Component', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const headerTitle = screen.getByText('Finance');
    expect(headerTitle).toBeInTheDocument();
  });

  it('Tests the display of Header Component', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tests the display of Navbar Component', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tests the display of Layout Component', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tests the display of Home Component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <HomeComponent />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tests the display of detail Component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Detail />
      </Provider>,

    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tests the display of Home page ', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tests the display of Detail page ', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Detail />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tests the display of App ', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
