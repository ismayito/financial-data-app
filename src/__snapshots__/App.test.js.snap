// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the financial App Tests the display of App  1`] = `
<div
  className="App"
>
  <div>
    <div
      className="links"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      />
    </div>
    <div
      className="homeContainer"
    >
      <div>
        <header>
          <nav
            className="navBar"
          >
            <section
              className="logo"
            >
              <button
                className="navigationButton"
                onClick={[Function]}
                type="submit"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  onClick={[Function]}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
                    fillRule="evenodd"
                  />
                </svg>
                <p
                  className="link"
                >
                  Finance
                </p>
              </button>
            </section>
            <p
              className="title"
            >
              Companies
            </p>
            <section
              className="navIcons"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
                />
              </svg>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
                />
              </svg>
            </section>
          </nav>
        </header>
      </div>
      <div
        className="headSection"
      >
        <section
          className="imageContainer"
        >
          <img
            alt="fruityImage"
            className="fruityImage"
            src="../assets/images/financeimage.webp"
          />
        </section>
        <section
          className="introInfo"
        >
          <h1
            className="heading"
          >
            Finance
            <br />
            <span>
              0
               
              companies
            </span>
          </h1>
        </section>
      </div>
      <section
        className="inputContainer"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="search..."
          type="search"
        />
      </section>
      <p
        className="priceTitle"
      >
        Price By Stock Company
      </p>
      <div
        className="cardContainer"
      />
    </div>
  </div>
</div>
`;

exports[`Tests for the financial App Tests the display of Detail page  1`] = `
<div>
  <div
    className="headSection"
  >
    <section
      className="imageContainer"
    >
      <img
        alt="financeImage"
        className="fruityImage"
        src="../assets/images/detail.jpg"
      />
    </section>
  </div>
</div>
`;

exports[`Tests for the financial App Tests the display of Header Component 1`] = `
<div>
  <header>
    <nav
      className="navBar"
    >
      <section
        className="logo"
      >
        <button
          className="navigationButton"
          onClick={[Function]}
          type="submit"
        >
          <svg
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
              fillRule="evenodd"
            />
          </svg>
          <p
            className="link"
          >
            Finance
          </p>
        </button>
      </section>
      <p
        className="title"
      />
      <section
        className="navIcons"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
          />
        </svg>
      </section>
    </nav>
  </header>
</div>
`;

exports[`Tests for the financial App Tests the display of Home Component 1`] = `
Array [
  <div
    className="headSection"
  >
    <section
      className="imageContainer"
    >
      <img
        alt="fruityImage"
        className="fruityImage"
        src="../assets/images/financeimage.webp"
      />
    </section>
    <section
      className="introInfo"
    >
      <h1
        className="heading"
      >
        Finance
        <br />
        <span>
          0
           
          companies
        </span>
      </h1>
    </section>
  </div>,
  <section
    className="inputContainer"
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="search..."
      type="search"
    />
  </section>,
  <p
    className="priceTitle"
  >
    Price By Stock Company
  </p>,
  <div
    className="cardContainer"
  />,
]
`;

exports[`Tests for the financial App Tests the display of Home page  1`] = `
<div
  className="homeContainer"
>
  <div>
    <header>
      <nav
        className="navBar"
      >
        <section
          className="logo"
        >
          <button
            className="navigationButton"
            onClick={[Function]}
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
                fillRule="evenodd"
              />
            </svg>
            <p
              className="link"
            >
              Finance
            </p>
          </button>
        </section>
        <p
          className="title"
        >
          Companies
        </p>
        <section
          className="navIcons"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
            />
          </svg>
        </section>
      </nav>
    </header>
  </div>
  <div
    className="headSection"
  >
    <section
      className="imageContainer"
    >
      <img
        alt="fruityImage"
        className="fruityImage"
        src="../assets/images/financeimage.webp"
      />
    </section>
    <section
      className="introInfo"
    >
      <h1
        className="heading"
      >
        Finance
        <br />
        <span>
          0
           
          companies
        </span>
      </h1>
    </section>
  </div>
  <section
    className="inputContainer"
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="search..."
      type="search"
    />
  </section>
  <p
    className="priceTitle"
  >
    Price By Stock Company
  </p>
  <div
    className="cardContainer"
  />
</div>
`;

exports[`Tests for the financial App Tests the display of Layout Component 1`] = `
<div>
  <div
    className="links"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Tests for the financial App Tests the display of Navbar Component 1`] = `
<div
  className="links"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  />
</div>
`;

exports[`Tests for the financial App Tests the display of detail Component 1`] = `
<div>
  <div
    className="headSection"
  >
    <section
      className="imageContainer"
    >
      <img
        alt="financeImage"
        className="fruityImage"
        src="../assets/images/detail.jpg"
      />
    </section>
  </div>
</div>
`;

exports[`Tests for the financial App Tests the finacialSlice  1`] = `
<div>
  <div
    className="headSection"
  >
    <section
      className="imageContainer"
    >
      <img
        alt="financeImage"
        className="fruityImage"
        src="../assets/images/detail.jpg"
      />
    </section>
  </div>
</div>
`;
